common/utils/Clock.cpp

```diff
@@ -253,6 +253,19 @@ const TimeStamp TimeStamp::operator-(const TimeInterval &interval) const {
 
 void Clock::CurrentTime(TimeStamp *timestamp) const {
   struct timeval tv;
+#ifdef CLOCK_MONOTONIC
+  struct timespec ts;
+  clock_gettime(CLOCK_MONOTONIC, &ts);
+  tv.tv_sec = ts.tv_sec;
+  tv.tv_usec = ts.tv_nsec / ONE_THOUSAND;
+#else
+  gettimeofday(&tv, NULL);
```

Comment:
I think it would be good to call the other wrapper function here (i.e. currently CurrentRealTime) rather than calling gettimeofday directly, in case we need to add other wrapping or ifdefs round it in future.

Reply:
I appreciate all the feedback, @peternewman! I'll start working on the requested changes.

> Rename CurrentTime to CurrentMonotonicTime (or similar)

I did consider this, but felt that if the function falls back to `gettimeofday`, then perhaps CurrentMonotonicTime is a bit of a misnomer.  I don't mind changing it either way, we'll just have to document the behavior.

Reply:
Agreed on documentation. I'd say the big benefit of renaming it is we can find stuff that hasn't been spotted (e.g. when this gets merged into our main branch there are new plugins, if we keep the same name, or if external people are using it, you can't tell if that code has been checked yet and deemed safe, or just missed).

I take your point about the fall back, do we know the likelyhood of it actually happening roughly (i.e. what doesn't support monotonic)? You do raise one good point, given it's "unexpected" we should log an info message when that path is called (assuming that's not too spammy; perhaps try a test run with it logging on both paths and running at `-l 4` logging). You could also add a warning to configure when it hasn't been found/supported.

Reply:
> I take your point about the fall back, do we know the likelyhood of it actually happening roughly (i.e. what doesn't support monotonic)?

Unsure of the likelihood, but I suspect "older" operating systems e.g. I believe it will fall back on OS X before 10.12 due to lack of `clock_gettime`.  

> we should log an info message when that path is called (assuming that's not too spammy; perhaps try a test run with it logging on both paths and running at -l 4 logging)

I tried logging these paths, unfortunately it is kind of noisy at about once per second so I removed that debug output.

> You could also add a warning to configure when it hasn't been found/supported.

I was able to add a configure check for clock_gettime, but was unsure of how to check that CLOCK_MONOTONIC is defined. Please let me know if you have any advice on this.

Reply:
> > we should log an info message when that path is called (assuming that's not too spammy; perhaps try a test run with it logging on both paths and running at -l 4 logging)
> 
> I tried logging these paths, unfortunately it is kind of noisy at about once per second so I removed that debug output.

Ah yeah, I guess that doubles the messages we get around the SelectServer stuff.

You could add something in the AppInit and ServerInit functions which calls InitClock to just log a message or something:
https://github.com/OpenLightingProject/ola/blob/d1010976c47de23dc6031749179f8d92596b7e8d/common/base/Init.cpp

Possibly OTT, although it would probably be nice to see the status when the program is run from a debugging point of view, rather than just a warning in configure. It could even print and display both clocks for comparison.

> > You could also add a warning to configure when it hasn't been found/supported.
> 
> I was able to add a configure check for clock_gettime, but was unsure of how to check that CLOCK_MONOTONIC is defined. Please let me know if you have any advice on this.

This would cover the warning:
https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Printing-Messages.html

And this the switching:
https://stackoverflow.com/questions/46704731/check-value-of-c-define-with-autoconf

To be honest I'm still torn, most people will probably ignore a configure warning if it succeeds, the app/server init would at least surface it in a log we might look at when they complain. Equally we've worked fine for the last however many years, so the risk of it causing an issue is hopefully fairly low either way.

Reply:
Now done.